plugin/trino-hive/src/main/java/io/trino/plugin/hive/CachingDirectoryLister.java

```diff
@@ -179,4 +214,48 @@ public long getRequestCount()
     {
         return cache.stats().requestCount();
     }
+
+    @VisibleForTesting
+    int getCachedPathsCount(String databaseName, String tableName)
+    {
+        SchemaTableName schemaTableName = new SchemaTableName(databaseName, tableName);
+        int count = 0;
+        for (TablePath cachedTablePath : cache.asMap().keySet()) {
+            if (cachedTablePath.schemaTableName.equals(schemaTableName)) {
+                count++;
+            }
+        }
+        return count;
+    }
+
+
+    private static class TablePath
+    {
+        private final SchemaTableName schemaTableName;
+        private final Path path;
+
+        public TablePath(SchemaTableName schemaTableName, Path path){
+            this.schemaTableName = schemaTableName;
+            this.path = path;
+        }
+
+        @Override
+        public boolean equals(Object o)
+        {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            TablePath cacheKey = (TablePath) o;
+            return Objects.equals(schemaTableName, cacheKey.schemaTableName) && Objects.equals(path, cacheKey.path);
```

Comment:
`Objects.equals` is nice, but redundant here

Reply:
We do have a `cache.getIfPresent` call in `io.trino.plugin.hive.CachingDirectoryLister#list` method.

Reply:
> Objects.equals is nice, but redundant here

Do you mean that we can just use `x.equals(other.x)` here, as those are not null. I think we generally do not care about that and use `Object.equals` as this is what intellij generates.

Reply:
> as this is what intellij generates.

depending on the checkboxes in the generation dialog

Reply:
Sorry. I misinterpreted your comment Piotr and thought you meant to drop the `#equals()` method altogether.
Sure I can simplify the implementation of the current method.